test/components/stage_progression_button_test.js

```diff
@@ -153,6 +153,53 @@ describe("StageProgressionButton", () => {
         expect(stageProgressionButton.find("div")).to.have.length(0)
       })
     })
+
+    context("when stateDependentTooltip returns a value", () => {
```

Comment:
@pwjablonski These tests are on point. Thank you for being thorough with the various cases handled.